### DISCUSSION TITLE
Resolution of gamepad axes

### DISCUSSION THREAD

**Discussion by LogicNoo** [hathach/tinyusb#1725] [Q&A]
Hi,
I'm trying to make a flight stick using Pi Pico. While in the process of butchering [hid_composite example](https://github.com/hathach/tinyusb/tree/master/examples/device/hid_composite) to my needs I have encountered the fact that it has only 8 bit resolution. 

I have tried changing descriptor `HID_LOGICAL_MIN` and `HID_LOGICAL_MAX`  in hid_devices.h and variable types to 12/16bit ones and but  getting  unsigned conversion error and my beginner C knowledge is not near enough to solve that.  Any help would be appreciated.

_Notes:_ changes made: 
  in _`hid_devices.h`_ lines197 - 202 & 155 & 88

  _`tud_hid_gamepad_report`_ and _`hid_gamepad_report_t`_ axes changed to _`int16_t`_ type. 

in _`hid.h`_  lines 329 & 330

_`HID_LOGICAL_MIN`_ to _`0x0801`_ and ` _HID_LOGICAL_MAX_`  to _`0x7ff`_

**Comment by hathach** [comment]
you need to declare 2 bytes values with HID_LOGICAL_MIN_N(u16, 2)
